LibZipSharp/Xamarin.Tools.Zip/ZipArchive.cs

```diff
@@ -773,34 +788,61 @@ internal static unsafe Int64 stream_callback (IntPtr state, IntPtr data, UInt64
 					return (Int64)sizeof (Native.zip_stat_t);
 
 				case SourceCommand.Tell:
-				case SourceCommand.TellWrite:
 					return (Int64)stream.Position;
+				case SourceCommand.TellWrite:
+					return (Int64)destination.Position;
 
 				case SourceCommand.Write:
 					buffer = ArrayPool<byte>.Shared.Rent (length);
 					try {
 						Marshal.Copy (data, buffer, 0, length);
-						stream.Write (buffer, 0, length);
+						destination.Write (buffer, 0, length);
 						return length;
 					} finally {
 						ArrayPool<byte>.Shared.Return (buffer);
 					}
 
 				case SourceCommand.SeekWrite:
-				case SourceCommand.Seek:
 					Native.zip_error_t error;
-					UInt64 offset = Native.zip_source_seek_compute_offset ((UInt64)stream.Position, (UInt64)stream.Length, data, len, out error);
-					stream.Seek ((long)offset, SeekOrigin.Begin);
+					Int64 offset = Native.zip_source_seek_compute_offset ((UInt64)destination.Position, (UInt64)destination.Length, data, len, out error);
+					if (offset < 0) {
+						return offset;
+					}
+					if (offset != stream.Seek (offset, SeekOrigin.Begin)) {
+						return -1;
+					}
+					break;
+				case SourceCommand.Seek:
+					offset = Native.zip_source_seek_compute_offset ((UInt64)stream.Position, (UInt64)stream.Length, data, len, out error);
+					if (offset < 0) {
+						return offset;
+					}
+					if (offset != stream.Seek (offset, SeekOrigin.Begin)) {
+						return -1;
+					}
 					break;
 
 				case SourceCommand.CommitWrite:
-					stream.Flush ();
+					destination.Flush ();
+					stream.Position = 0;
+					destination.Position = 0;
+					stream.SetLength (destination.Length);
+					destination.CopyTo (stream);
+					stream.Position = 0;
+					destination.Dispose ();
+					context.Destination = null;
+					if (!string.IsNullOrEmpty (context.DestinationFileName)&& File.Exists (context.DestinationFileName)) {
+						File.Delete (context.DestinationFileName);
```

Comment:
Maybe catch and ignore (after writing a warning) any exceptions thrown by `File.Delete`? If the temporary file is left behind, too bad, but I think it's not worth aborting the process in such a case.

Reply:
should we just write a warning using `Console.WriteLine` ? or stick with `Debug.WriteLine` ? I'm not sure what is best in this senario cos a) there isn't much the user can do about it and b) LibZipSharp doesn't really have a warning reporting mechanism.

Reply:
Can we use both? It will show up **somewhere** - it's mostly for us, should we need to diagnose something. I just dislike `catch` which hide exceptions :)